common-custom-user-data-gradle-plugin/src/main/java/com/gradle/CustomBuildScanEnhancements.java

```diff
@@ -239,9 +239,17 @@ private void captureGitMetadata() {
             String gitRepo = execAndGetStdOut("git", "config", "--get", "remote.origin.url");
             String gitCommitId = execAndGetStdOut("git", "rev-parse", "--verify", "HEAD");
             String gitCommitShortId = execAndGetStdOut("git", "rev-parse", "--short=8", "--verify", "HEAD");
-            String gitBranchName = execAndGetStdOut("git", "rev-parse", "--abbrev-ref", "HEAD");
             String gitStatus = execAndGetStdOut("git", "status", "--porcelain");
 
+            String gitBranchName;
```

Comment:
I would replace this with 

`String gitBranchName = ((isJenkins() || isHudson()) && envVariable("BRANCH_NAME).isPresent()) ? envVariable("BRANCH_NAME).get() : execAndGetStdOut("git", "rev-parse", "--abbrev-ref", "HEAD");`

Reply:
I thought about doing that, but then the isJenkins and isHudson functions will need to be made static, and then we'll need to be able to pass in the Providers object. Which we certainly could do, but we will loose out on some of the benefits from the earlier refactoring.

Reply:
I have some ideas on how to do this and still preserve the benefits of the refactoring. I'll gave them a try and will update the PR accordingly.

Reply:
I just added a commit that attempts to resolve this. It adds a small bit of duplication, but I think it makes a few things clearer/more obvious, while also keeping the diff between the Gradle and Maven versions of the file as similar as possible.